mod.d/kpti.yaml

```diff
@@ -0,0 +1,160 @@
+
+      Returns:
+          True (bool): If using an Intel CPU
+          False (bool): If using an AMD CPU
+      """
+      try:
+          with open("/proc/cpuinfo", "r") as cpuinfo_file:
+              if re.search("GenuineIntel", cpuinfo_file.read()):
+                  print("Running on an Intel CPU.\n")
+                  return True
+              elif re.search("AuthenticAMD", cpuinfo_file.read()):
+                  print("Running on an AMD CPU.\n")
+                  return False
+              else:
+                  print("Running on a non-Intel or AMD CPU. This is currently unsupported.\n")
+                  sys.exit(0)
+      except Exception as ex:
+          print(ex)
+          print("[WARN] Unable to determine CPU vendor\n")
+          raise
+
+
+  def get_kpti_status():
+      """
+      Determines if KPTI is in use.
+
+      Returns:
+         True (bool): If KPTI is enabled
+      """
+      try:
+          with open("/sys/devices/system/cpu/vulnerabilities/meltdown", "r") as meltdown_file:
+              meltdown = meltdown_file.read()
+              print(meltdown)
+              if re.search("PTI", meltdown):
+                  print("Kernel Page Table Isolation is enabled\n")
+                  return True
+              else:
+                  print("Kernel Page Table Isolation is disabled\n")
+                  with open("/proc/cmdline", "r") as cmdline_file:
+                      if re.search("nopti", cmdline_file.read()):
+                          print("Kernel Page Table isolation is disabled via nopti on the kernel commandline.\n")
+                  return False
+      except Exception as ex:
+          print(ex)
+          print("[WARN] Unable to determine KPTI status\n")
+          raise
+
+  def get_pcid_status():
+      """
+      Determine if the system supports pcid or nvpcid. Pre-Haswell CPUs should only have PCID support.
+      """
+      try:
+          with open("/proc/cpuinfo", "r") as cpuinfo_file:
+              if re.search("invpcid", cpuinfo_file.read()):
+                  print("invpcid supported. This is the best case for Kernel Page Table Isolation performance.\n")
+              elif re.search("pcid", cpuinfo_file.read()):
+                  print("pcid supported. This is a moderate case for Kernel Page Table Isolation performance.\n")
+              else:
+                  print("No pcid support detected. This is the worst case for Kernel Page Table Isolation performance.\n")
+      except Exception as ex:
+          print(ex)
+          print("[WARN] Unable to determine pcid support\n")
+          raise
+
+  def run():
+      """
+      Determines if KPTI is enabled and what level of pcid support is provided.
+      Returns:
+          True (bool): If collection is successful.
+      """
+      try:
+          kpti = False
+          if is_intel_cpu():
+              print("Continuing on to determine Meltdown related status\n")
+              if get_kpti_status():
```

Comment:
Since get_kpti_status returns a boolean, you could replace the if block with a single line:
`kpti = get_kpti_status()`